test/sys/test_pthread.rs

```diff
@@ -20,3 +20,18 @@ fn test_pthread_kill_none() {
     pthread_kill(pthread_self(), None)
         .expect("Should be able to send signal to my thread.");
 }
+
+#[test]
+#[cfg(all(
+    any(target_os = "linux", target_os = "android"),
+    target_env = "gnu"
+))]
+fn test_pthread_sigqueue_none() {
+    use std::ptr::null_mut;
```

Comment:
So this is actually sending a signal?  If so, you must lock SIGNAL_MTX.

Reply:
This doesn't actually send a signal. The signal that ends being passed to the syscall is signal 0, which is a special case that doesn't actually send the signal(see: https://stackoverflow.com/questions/32259228/what-is-the-zero-signal).

I basically did the same thing as the test for fill above.